ceph/rados/serviceability_workflows.py

```diff
@@ -225,22 +225,71 @@ def remove_custom_host(self, host_node_name: str):
         Returns:
             None | raises exception in case of failure
         """
+        status_cmd = ""
         try:
+
+            def wait_osd_operation_status(status_cmd):
+                status_flag = False
+                end_time = datetime.datetime.now() + datetime.timedelta(seconds=600)
+                while end_time > datetime.datetime.now():
+                    out, err = self.cephadm.shell([status_cmd])
+                    try:
+                        drain_ops = json.loads(out)
+                        for entry in drain_ops:
+                            log.debug(
+                                f"OSD remove operation is in progress {osd_id}\nOperations: {entry}"
+                            )
+                    except json.JSONDecodeError:
+                        log.info(f"The OSD removal is completed on OSD : {osd_id}")
```

Comment:
This could either mean that OSD removal is complete, or No OSD removal was started in the 1st place.

Reply:
We could use `ceph pg ls-by-osd OSD_ID` to check if PGs are drained or not.

Reply:
Implemented the logic by using the ceph orch ps hostname command